pkg/auth/providers/ldap/ldap_client.go

```diff
@@ -140,6 +141,7 @@ func (p *ldapProvider) getPrincipalsFromSearchResult(result *ldapv2.SearchResult
 	}
 
 	userPrincipal = *user
+	userDN := result.Entries[0].DN
```

Comment:
can entries be ever empty? :)

Reply:
it can't be because we ensure in the methods `loginUser` and `RefetchGroupPrincipals` that call `getPrincipalsFromSearchResult` that len(result.Entries) = 1